### ISSUE TITLE
Project Find and Replace hangs indefinitely on projects containing symlinks

### ISSUE THREAD

**Issue by Beefster09** [atom/atom#16326]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

If you try to do a Project Find and Replace on a project that contains symlinks, the Find and Replace will hang indefinitely (probably forever) with empty or incomplete results.

### Steps to Reproduce

1. Create a folder containing a file of any type
2. Create a relative symlink to that file (I used WSL/Ubuntu bash: `ln -rs file symlink`)
3. Open the Project Find and Replace window and search for any string. It will hang forever on the symlink.

**Expected behavior:** [What you expect to happen]
Search completes as usual, symlinks ignored

**Actual behavior:** [What actually happens]
Search hangs at first symlink encountered

**Reproduces how often:** [What percentage of the time does it reproduce?]
Every time

### Versions

Windows 10 Enterprise 1703, 64 bit
HP EliteBook
i5-6300U @ 2.4 GHz
8 GB RAM

atom --version
> Atom    : 1.22.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm --version
> apm  1.18.8
npm  3.10.10
node 6.9.5 x64
python 2.7.13
git
visual studio

### Additional Information

This might be Windows-only due to how Git for Windows/WSL handles symlinks.
The folder is native Windows.

**Comment by rsese** [comment]
Thanks for the report!

> This might be Windows-only due to how Git for Windows/WSL handles symlinks.

Hmmm, yeah I think this is related specifically to WSL symlinks?  I couldn't reproduce this behavior on macOS or Ubuntu, or on Windows 10 using `mklink` from cmd.exe.

I noticed that the link doesn't show up in the tree view like the other links I created and I did see an error in developer tools console.  The error led me to:

https://github.com/Microsoft/WSL/issues/353

So it seems like an issue with the links created in WSL?
